pkgs/by-name/li/libnvidia-container/fix-library-resolving.patch

```diff
@@ -52,14 +56,25 @@ index 38bab05..e1abc89 100644
 -                        }
 -                        break;
 -                }
++        for (size_t j = 0; j < size; ++j) {
++                snprintf(dir, 100, "/run/opengl-driver%s/lib",
++                    arch == LD_I386_LIB32 ? "-32" : "");
```

Comment:
(addressing this comment is not necessary for merging)

A note on getting up-to-date: `addOpenGLRunpath` has been renamed to `addDriverRunpath` as part of the transition to opengl-agnostic names; this bit would be `${addDriverRunpath.driverLink}/lib`.

I'm not sure if we need to keep the 32-bit detection logic because:

```
❯ nix eval nixpkgs#addDriverRunpath.driverLink
"/run/opengl-driver" 
❯ nix eval nixpkgs#pkgsCross.musl32.addDriverRunpath.driverLink
"/run/opengl-driver-32"
```

Reply:
So would we rather parametrize the patch file here than deciding at runtime?

Reply:
Yes, a common pattern is to use `substituteAll`